### ISSUE TITLE
Icons not accessible in Safari + VoiceOver

### ISSUE THREAD

**Issue by Vanals** [emotion-icons/emotion-icons#172]
NPM package: @emotion-icons

Hi!

In one of the projects I am working for, I have introduced the material emotion icons. They worked very well, and I've tested their functionality, accessibility included.

Initially, I tested the functionality only in Chrome, where worked well but then I've realized that the icons **are not accessible in Safari**.

The way you are making the icons accessible is by setting aria-hidden='false' in the < svg > element and then passing a title inside the < svg >. 
The < title> contains the text the screen reader will read.

Unfortunately, this is not enough for Safari. In order to work in Safari, the < svg > element should also include an 
aria-labelledby attribute and the title should have an id which value has to be the same of the aria-labelledby; used by the < svg >.

```
<svg aria-labelledby="ok1">
  <title id='ok1'>text read by sr</title>
</svg>

```
**The id has to be unique of course, to avoid clashes if more instances of the same icon are implemented in the same page.**
The accessibility of the other "browser + screen reader" combination should not be compromised at all.

What do you think? Hope this can be implemented?

Thanks

**Comment by Vanals** [comment]
I wrongly only checked the icon's accessibility inside a button tag, in that case, the button was not accessible in Safari, using the technique you are currently using. But the icon in isolation it is.  

The solution would be work on the Button element itself rather than the svg

**Comment by jacobwgillespie** [comment]
Hey, thanks for the report!

Just so I'm clear, you're saying that if you wrap an icon in a `<button>`, you need to apply the title to the button, not the icon itself? If so, is that preferable, that the button is labeled rather than the icon?

If not, or if you have any other accessibility suggestions, definitely let me know and I'd be happy to look into implementation.

**Comment by Vanals** [comment]
> Hey, thanks for the report!
> 
> Just so I'm clear, you're saying that if you wrap an icon in a `<button>`, you need to apply the title to the button, not the icon itself? If so, is that preferable, that the button is labeled rather than the icon?
> 
> If not, or if you have any other accessibility suggestions, definitely let me know and I'd be happy to look into implementation.

Exactly! The < title > tag inside the < svg > won't work for Safari + VoiceOver. But works on Chrome for example. 
If you wrap the material Icons in a button it won't make the button accessible in Safari + VoiceOver. The svg  < title > won't be read. 

To make an icon button accessible, using material icons and not acting on the <button > tag, you would need to follow the technique I have explained in my main post. But is probably more complex as the id has to be unique.

You can instead act directly on the <button > tag. It is completely fine to pass an aria-label to the < button >, for example. It is a good practice.
I believe no change is required :).

Initially, I thought the svg icon was not accessible in safari + voice over at all. But actually with the implementation currently set, the icon in isolation is read properly by the SR.

Hope make sense!  👍

**Comment by jacobwgillespie** [comment]
Cool, yeah thanks for the detail, this makes sense. I'll close this issue for now, but definitely post back if it turns out that we do need to adjust the icon accessibility props 👍
